### PR TITLE
feat: t2006 files added

### PR DESCRIPTION
**Pull request by Abishek666**
Task 6 - API ADDED

### CODE REVIEW

File: src/baseClass.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { topSection } from './topSection.js'
```
Comment (saikrishnanp) [inline src/baseClass.js:1]: Continent name wise sorting doesn't seem to be clickable now, could you fix that?

File: src/fetchData.js [hunk 2]
```diff
@@ -4,9 +4,42 @@ import { sortElements } from './sortElements.js'
  * @returns {object} cityDetails
  */
 export async function fetchCityDetails () {
-  let cityDetails = await fetch('../Assets/HTML&CSS/files/data.json')
+  let cityDetails = await fetch('https://soliton.glitch.me/all-timezone-cities')
     .then((res) => { return res.json() })
```
Comment (saikrishnanp) [inline src/fetchData.js]: Do we need .then blocks when await is used?

File: src/fetchData.js [hunk 3]
```diff
@@ -4,9 +4,42 @@ import { sortElements } from './sortElements.js'
  * @returns {object} cityDetails
  */
 export async function fetchCityDetails () {
-  let cityDetails = await fetch('../Assets/HTML&CSS/files/data.json')
+  let cityDetails = await fetch('https://soliton.glitch.me/all-timezone-cities')
```
Comment (saikrishnanp) [inline src/fetchData.js]: Can we have the url stored in some const?

File: src/fetchData.js [hunk 4]
```diff
@@ -4,9 +4,42 @@ import { sortElements } from './sortElements.js'
...
     .then((res) => { return res.json() })
     .then((data) => { return data })
+  const cityDetailsObject = {}
+  cityDetails.map((items) => {
```
Comment (saikrishnanp) [inline src/fetchData.js]: Check if forEach can be used in case we don't need a new array to be created

File: src/middleSection.js [hunk 5]
```diff
@@ -89,8 +96,32 @@ const changeMiddleSegment = (cities) => {
...
   updateScrollButtonsVisibility()
 }
 
-export const middleSection = { updateScrollButtonsVisibility, addCardsbySpinner, scrollItems, selectIcon, changeMiddleSegment, filterMiddleSegment }
+/**
+Method to scroll Middle section both Horizontally and Vertically
+ * @param {object} target - target element
+ * @param {object} targetContainer - underlying container
+ */
+function yScroll (target, targetContainer) {
```
Comment (saikrishnanp) [inline src/middleSection.js]: function name could be more meaningful

File: src/time.js [hunk 6]
```diff
@@ -28,13 +28,28 @@ const updateCityTileTime = (cityTimezone, itemName) => {
   }, 60000)
 }
 
+/**
+ * This function will get the current date
+ * @param {string} cityTimezone Timezone name
+ * @returns  {Array} returns the date array
+ */
+
+const updateDate = (cityTimezone) => {
```
Comment (saikrishnanp) [inline src/time.js]: As this function returns and not updates
```suggestion
const getDate = (cityTimezone) => {
```

**Review by saikrishnanp — CHANGES_REQUESTED** [review]

File: src/fetchData.js (hunk 2 above)
Comment (Abishek666) [reply src/fetchData.js]: no Sai, either  we can use either one...i will change it

**Pull request closed without merging**